src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -476,19 +476,28 @@ internal bool IsEncodingInRequestAcceptEncoding(string encoding)
                 while (true)
                 {
                     var separatorIndex = acceptEncoding.IndexOf(',');
-                    if (separatorIndex == -1)
```

Comment:
Use the typed header? cc @Tratcher

Reply:
The header is custom to gRPC: `grpc-accept-encoding`. However if there is a better way of reading a comma delimited list of values from a header then that could replace this code

Reply:
Like this? https://github.com/aspnet/AspNetCore/blob/4ef204e13b88c0734e0e94a1cc4c0ef05f40849e/src/Http/Http.Abstractions/src/Extensions/HeaderDictionaryExtensions.cs#L39

Reply:
Allocations 😲

Reply:
I know right... 😄 We need to expose another API for the split. 

Unrelated: There's a new API designed by a corefx contributor that beautifully handles splitting things by returning an enumerator of ranges.